### PR TITLE
NITF: Enables reading of new TRE field types

### PR DESCRIPTION
**Pull request by crossbearer**
A new version of NITF will be available soon (definition file attached from https://nsgreg.nga.mil/doc/view?i=4829). This file has TREs with field value types of IEEE754 Float and UINT as opposed to the traditional string parsing.
[NGA.STND.0072_1.0_SNIP_Version_1.0_07-June-2019(1).pdf](https://github.com/OSGeo/gdal/files/4360387/NGA.STND.0072_1.0_SNIP_Version_1.0_07-June-2019.1.pdf)

### CODE REVIEW

File: gdal/frmts/nitf/nitfdataset.cpp [hunk 1]
```diff
@@ -2281,7 +2281,7 @@ CPLErr NITFDataset::_SetProjection(const char* _pszProjection)
     return CE_None;
 }
 
-#ifdef ESRI_BUILD
+//#ifdef ESRI_BUILD
```
Comment (rouault) [inline gdal/frmts/nitf/nitfdataset.cpp]: why are those // comments needed ? This is already protected by #ifdef

File: gdal/frmts/nitf/nitffile.c [hunk 2]
```diff
@@ -1,2928 +1,3001 @@
...
-    const double b = 6356752.3142;
-
-    /* convert to radians */
-    dfLat = dfLat * M_PI / 180.0;
-
-    /* convert to geodetic */
-    dfLat = atan( ((a*a)/(b*b)) * tan(dfLat) );
-
-    /* convert back to degrees */
-
```
Comment (rouault) [inline gdal/frmts/nitf/nitffile.c]: I suspect there must be some newline changes, or perhaps just the change of the file permissions, that cause the difference to appear to be on the whole file and makes it unreviewable

File: gdal/configure.ac [hunk 3]
```diff
@@ -111,7 +111,7 @@ AX_CHECK_COMPILE_FLAG([-Wdocumentation -Wno-documentation-deprecated-sync], [C_W
 dnl gnu89 is a reasonable target to get MSVC compatibility.
 dnl but only apply it with gcc, since clang will throw a lot of warnings
 SAVED_CFLAGS=$CFLAGS
-CFLAGS="$CFLAGS $ERROR_ON_UNKNOWN_OPTIONS -std=gnu89"
+CFLAGS="$CFLAGS $ERROR_ON_UNKNOWN_OPTIONS -std=c11"
```
Comment (rouault) [inline gdal/configure.ac]: the above comment about gnu89 should be changed

Comment (crossbearer) [reply gdal/configure.ac]: Reverted

File: gdal/frmts/nitf/nitfdataset.cpp (hunk 1 above)
Comment (crossbearer) [reply gdal/frmts/nitf/nitfdataset.cpp]: Reverted

**Pull request closed without merging**